kornia/feature/hynet.py

```diff
@@ -0,0 +1,219 @@
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return torch.max(x, self.tau)
+
+
+class HyNet(nn.Module):
+    r"""Module, which computes HyNet descriptors of given grayscale patches of 32x32.
+
+    This is based on the original code from paper
+    "HyNet: Learning Local Descriptor with Hybrid Similarity Measure and Triplet Loss".
+    See :cite:`hynet2020` for more details.
+
+    Args:
+        pretrained: Download and set pretrained weights to the model.
+        is_bias: use bias in TLU layers
+        is_bias_FRN:  use bias in FRN layers
+        dim_desc: descriptor dimentionality,
+        drop_rate: dropout rate,
+        eps_l2_norm: to avoid div by zero
+
+    Returns:
+        torch.Tensor: HyNet descriptor of the patches.
+
+    Shape:
+        - Input: :math:`(B, 1, 32, 32)`
+        - Output: :math:`(B, 128)`
+
+    Examples:
+        >>> input = torch.rand(16, 1, 32, 32)
+        >>> hynet = HyNet()
+        >>> descs = hynet(input) # 16x128
+    """
+    patch_size = 32
+
+    def __init__(self, pretrained: bool = False,
+                 is_bias: bool = True,
+                 is_bias_FRN: bool = True,
+                 dim_desc: int = 128,
+                 drop_rate: float = 0.3,
+                 eps_l2_norm: float = 1e-10):
+        super().__init__()
+        self.eps_l2_norm = eps_l2_norm
+        self.dim_desc = dim_desc
+        self.drop_rate = drop_rate
+        self.layer1 = nn.Sequential(
+            FilterResponseNorm2d(1, is_bias=is_bias_FRN),
+            TLU(1),
+            nn.Conv2d(1, 32, kernel_size=3, padding=1, bias=is_bias),
+            FilterResponseNorm2d(32, is_bias=is_bias_FRN),
+            TLU(32),
+        )
+
+        self.layer2 = nn.Sequential(
+            nn.Conv2d(32, 32, kernel_size=3, padding=1, bias=is_bias),
+            FilterResponseNorm2d(32, is_bias=is_bias_FRN),
+            TLU(32),
+        )
+
+        self.layer3 = nn.Sequential(
+            nn.Conv2d(32, 64, kernel_size=3, stride=2, padding=1, bias=is_bias),
+            FilterResponseNorm2d(64, is_bias=is_bias_FRN),
+            TLU(64),
+        )
+
+        self.layer4 = nn.Sequential(
+            nn.Conv2d(64, 64, kernel_size=3, padding=1, bias=is_bias),
+            FilterResponseNorm2d(64, is_bias=is_bias_FRN),
+            TLU(64),
+        )
+        self.layer5 = nn.Sequential(
+            nn.Conv2d(64, 128, kernel_size=3, stride=2, padding=1, bias=is_bias),
+            FilterResponseNorm2d(128, is_bias=is_bias_FRN),
+            TLU(128),
+        )
+
+        self.layer6 = nn.Sequential(
+            nn.Conv2d(128, 128, kernel_size=3, padding=1, bias=is_bias),
+            FilterResponseNorm2d(128, is_bias=is_bias_FRN),
+            TLU(128),
+        )
+
+        self.layer7 = nn.Sequential(
+            nn.Dropout(self.drop_rate),
+            nn.Conv2d(128, self.dim_desc, kernel_size=8, bias=False),
+            nn.BatchNorm2d(self.dim_desc, affine=False)
+        )
```

Comment:
```suggestion
        )
        self.features = nn.Sequential([layer1, …])
```
I believe torch script later doesn’t like iterating over that list